ldap/servers/slapd/conntable.c

```diff
@@ -129,20 +129,21 @@ connection_table_new(int table_size)
     int free_idx = 0;
     ber_len_t maxbersize = config_get_maxbersize();
     ct = (Connection_Table *)slapi_ch_calloc(1, sizeof(Connection_Table));
-    ct->size = table_size;
-    ct->list_num = SLAPD_DEFAULT_NUM_LISTENERS;
-    ct->list_size = (table_size+ct->list_num-1)/ct->list_num; /* to avoid rounding issue */
     ct->num_active = (int *)slapi_ch_calloc(1, ct->list_num * sizeof(int));
-    ct->c = (Connection **)slapi_ch_calloc(1, table_size * sizeof(Connection *));
+    ct->list_num = config_get_num_listeners();
+    ct->size = table_size - (table_size % ct->list_num);
+    ct->list_size = ct->size/ct->list_num;
+    ct->num_active = (int *)slapi_ch_calloc(1, ct->list_num * sizeof(int));
+    ct->c = (Connection **)slapi_ch_calloc(1, ct->size * sizeof(Connection *));
     ct->fd = (struct POLL_STRUCT **)slapi_ch_calloc(1, ct->list_num * sizeof(struct POLL_STRUCT*));
     ct->table_mutex = PR_NewLock();
     /* Allocate the freelist */
-    ct->c_freelist = (Connection **)slapi_ch_calloc(1, table_size * sizeof(Connection *));
+    ct->c_freelist = (Connection **)slapi_ch_calloc(1, ct->size * sizeof(Connection *));
     /* NEVER use slot 0, this is a list pointer */
     ct->conn_next_offset = 1;
     ct->conn_free_offset = 1;
 
-    slapi_log_err(SLAPI_LOG_INFO, "connection_table_new", "conntablesize:%d\n", ct->size);
+    slapi_log_err(SLAPI_LOG_INFO, "connection_table_new", "list_num:%d - list_size:%d\n",  ct->list_num, ct->list_size);
```

Comment:
This message is logged systematically. It looks a bit  too internals for the admin. The admin set up N listeners, you may log that connection table is split in N sub-tables each containing M slots.